python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -161,6 +161,26 @@ def get_downstream_partitions_for_partitions(
             current_time=current_time,
         ).partitions_subset
 
+    def _merge_time_windows(self, time_windows: Sequence[TimeWindow]) -> Sequence[TimeWindow]:
+        """Takes a list of potentially-overlapping TimeWindows and merges any overlapping windows."""
+        if not time_windows:
+            return []
+
+        sorted_time_windows = sorted(time_windows, key=lambda tw: tw.start.timestamp())
+        merged_time_windows: List[TimeWindow] = [sorted_time_windows[0]]
+
+        for window in sorted_time_windows[1:]:
+            last_window = merged_time_windows[-1]
+            # window starts before the end of the previous one
+            if window.start.timestamp() <= last_window.end.timestamp():
+                merged_time_windows[-1] = TimeWindow(
+                    last_window.start, max(last_window.end, window.end, key=lambda d: d.timestamp())
```

Comment:
these timestamp() calls are unfortunately necessary even though datetimes support comparison operations due to the cursed knowledge that these comparison operators sometimes return the wrong answer when dealing with DST transitions :|

Reply:
dst strikes again